packages/devtools_app/lib/src/screens/memory/primitives/class_name.dart

```diff
@@ -0,0 +1,265 @@
+
+/// List of packages, published by Dart and Flutter.
+///
+/// There is no active monitoring for new packages.
+/// So, if you see something is missing here,
+/// please, create PR to add it.
+/// TODO(polina-c): add a test that verifies there is missing
+/// packages.
+@immutable
+class HeapClassName {
+  HeapClassName({required this.className, required this.library}) {
+    assert(
+      !isCore || !isDartOrFlutter,
+      'isCore and isDartOrFlutter must be exclusive',
+    );
+  }
+
+  HeapClassName.fromClassRef(ClassRef? classRef)
+      : library = _library(classRef?.library?.name, classRef?.library?.uri),
+        className = classRef?.name ?? '';
+
+  HeapClassName.fromHeapSnapshotClass(HeapSnapshotClass? theClass)
+      : library =
+            _library(theClass?.libraryName, theClass?.libraryUri.toString()),
+        className = theClass?.name ?? '';
+
+  static String _library(String? libName, String? libUrl) {
+    libName ??= '';
+    if (libName.isNotEmpty) return libName;
+    return libUrl ?? '';
+  }
+
+  final String className;
+  final String library;
+
+  String get fullName => library.isNotEmpty ? '$library/$className' : className;
+
+  bool get isSentinel => className == 'Sentinel' && library.isEmpty;
+
+  /// Detects if a class can retain an object from garbage collection.
+  bool get isWeakEntry {
+    // Classes that hold reference to an object without preventing
+    // its collection.
+    const weakHolders = {
+      '_WeakProperty': 'dart.core',
+      '_WeakReferenceImpl': 'dart.core',
+      'FinalizerEntry': 'dart._internal',
+    };
+
+    if (!weakHolders.containsKey(className)) return false;
+    if (weakHolders[className] == library) return true;
+
+    // If a class lives in unexpected library, this can be because of
+    // (1) name collision or (2) bug in this code.
+    // Throwing exception in debug mode to verify option #2.
+    // TODO(polina-c): create a way for users to add their weak classes
+    // or detect weak references automatically, without hard coding
+    // class names.
+    assert(false, 'Unexpected library for $className: $library.');
+    return false;
+  }
+
+  /// True, if the library is a core library.
+  ///
+  /// I.e. if the library name is empty or does not have prefix
+  /// `dart.` or `package:`.
+  bool get isCore =>
+      library.isEmpty ||
+      (!library.startsWith('dart.') && !library.startsWith('package:'));
+
+  /// True, if the package has prefix `dart.` or has perfix `package:` and is
+  /// published by Dart or Flutter org.
+  bool get isDartOrFlutter {
+    if (library.startsWith('dart.')) return true;
+
+    const packagePrefix = 'package:';
+    if (!library.startsWith(packagePrefix)) return false;
+    final slashIndex = library.indexOf('/');
+    if (slashIndex == -1) return false;
+    final packageName = library.substring(packagePrefix.length, slashIndex);
+
+    return _dartAndFlutterPackages.contains(packageName);
+  }
```

Comment:
pull the prefixes from here into a shared location and use in both places: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart#L629-L639